File: totosuki/ABC/A/278.py
N, K = map(int, input().split())
A = list(map(int, input().split()))
rslt = [a for a in A]
rslt = rslt[K:]
for _ in range(K):
  if len(rslt) < len(A):
    rslt.append(0)
print(*rslt)

File: OCH_Tao/ABC/A/278.js
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"));
const [, K] = input[0].split(" ").map(x => parseInt(x, 10));
const A = input[1].split(" ");
for (let i = 0; i < K; i++) {
  A.shift();
  A.push("0");
}
console.log(A.join(" "));